tests/phpunit/CRM/Case/BAO/CaseTypeTest.php

```diff
@@ -142,6 +142,41 @@ public function testConvertXmlToDefinition($fixtureName, $expectedJson, $inputXm
     $this->assertEquals($expectedDefinition, $actualDefinition);
   }
 
+  /**
+   * @param string $fixtureName
+   * @param string $expectedJson
+   * @param string $inputXml
+   * @dataProvider definitionProvider
+   */
+  public function testUpdateCaseRoleName($fixtureName, $expectedJson, $inputXml) {
+    if ($fixtureName == 'one-item-in-each') {
+      $xml = simplexml_load_string($inputXml);
+      $actualDefinition = CRM_Case_BAO_CaseType::convertXmlToDefinition($xml);
+      $relType = $this->callAPISuccess('RelationshipType', 'create', [
+        'name_a_b' => "First role",
+        'name_b_a' => "First role",
```

Comment:
Suggesting this change so that this test could also catch situations where a_b vs b_a has got mixed up.
```suggestion
        'name_b_a' => "First role is",
```

Also am I understanding the proposed fix is that when a relationship type **"label"** changes, this updates the **"name"** field in the xml definition?

Reply:
ok, I now see it is the label(not name) which is stored in XML definition. So not completely sure if we need to store name instead of label in case_type table or proceeding this way could be acceptable. Hmm. Thoughts @eileenmcnaughton @colemanw